cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -656,20 +655,15 @@ constructTxBodyContent sbe mScriptValidity mPparams inputsAndMaybeScriptWits rea
 
   validatedCollateralTxIns <- validateTxInsCollateral sbe txinsc
   validatedRefInputs <- validateTxInsReference sbe allReferenceInputs
-  validatedTotCollateral
-    <- first TxCmdTotalCollateralValidationError $ validateTxTotalCollateral era mTotCollateral
-  validatedRetCol
-    <- first TxCmdReturnCollateralValidationError $ validateTxReturnCollateral era mReturnCollateral
+  validatedTotCollateral <- first TxCmdNotSupportedInEraValidationError $ validateTxTotalCollateral sbe mTotCollateral
+  validatedRetCol        <- first TxCmdNotSupportedInEraValidationError $ validateTxReturnCollateral sbe mReturnCollateral
   let txFee = TxFeeExplicit sbe fee
-  validatedLowerBound <- first TxCmdTxValidityLowerBoundValidationError (validateTxValidityLowerBound era mLowerBound)
-  validatedReqSigners <- first TxCmdRequiredSignersValidationError $ validateRequiredSigners era reqSigners
-  validatedTxWtdrwls <- first TxCmdTxWithdrawalsValidationError $ validateTxWithdrawals era withdrawals
-  validatedTxCerts <- first TxCmdTxCertificatesValidationError $ validateTxCertificates era certsAndMaybeScriptWits
+  validatedLowerBound <- first TxCmdNotSupportedInEraValidationError $ validateTxValidityLowerBound sbe mLowerBound
```

Comment:
@Jimbo4350> not sure what you want here. In the current state of this PR, `validateTxValidityLowerBound` takes a `ShelleyBasedEra era` witness (this is an improvement, it was taking `CardanoEra era` before this PR).

If I tighten its type to mimick the API one, for example requiring `AllegraEraOnwards era`, we are going to push the era's verification to the caller (i.e. in `Transaction.hs`) and I think the point of the current design was to have verifications done in `TxValidationErrors.hs`.

What I could do is to remove the call to `conjureWitness` and use [caseShelleyEraOnlyOrAllegraEraOnwards](https://github.com/IntersectMBO/cardano-api/blob/smelc/more-logging-in-evaluateTransactionExecutionUnit/cardano-api/internal/Cardano/Api/Eras/Case.hs#L80). This would make the code more idiomatic, since that's what we do in more modern parts of the codebase. Is that the change you want?

Reply:
I was on the wrong branch :man_facepalming:. This is fine as is.